### ISSUE TITLE
Pdoc exits with 0 although documentation generation partially failed

### ISSUE THREAD

**Issue by lovetheguitar** [mitmproxy/pdoc#407]
#### Problem Description
When documenting a package and a module fails to be documented because of a missing/broken import, pdoc exits with `0`.
IMHO pdoc should exit `!= 0`, although partial documentation is generated.

#### Setup
```shell
pdoc_poc/
├── __init__.py
├── pdoc_fail.py
└── pdoc_success.py
```
`__init__.py` (empty)

`pdoc_fail.py`
```python
"""This aint gonna work..."""

import non_existant_module
```

`pdoc_success.py`
```python
"""
Thanks for developing pdoc... its really cool!
"""
```

1. call `pdoc pdoc_poc -o poc_html` 
2. check error code `echo $LASTEXITCODE` -> 0 

#### System Information
pdoc: 12.0.1
Python: 3.8.10
Platform: Windows-10-10.0.19042-SP0

**Comment by mhils** [comment]
Thanks for raising this and sorry for the slow reply! :)

I think I agree that silently swallowing an error isn't good here, we should get this fixed. In the meantime, you can specify `PYTHONWARNINGS` as a workaround:

```
PYTHONWARNINGS=error pdoc pdoc_poc -o poc_html
```

Hope this helps! :)

**Comment by mhils** [comment]
This has been fixed in cdce38f. :)
